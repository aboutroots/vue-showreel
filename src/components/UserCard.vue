<template>
  <router-link :to="{ name: 'User Detail', params: { id: user.id } }" class="user-card">
    <img :src="user.avatar" class="avatar" alt="User avatar" />
    <h3 class="title">{{ fullName }}</h3>
    <p class="subtitle">{{ user.email }}</p>
  </router-link>
</template>

<script lang="ts">
import { UserDTO } from '@/models';
import Vue, { PropType } from 'vue';

export default Vue.extend({
  name: 'UserCard',
  props: {
    user: {
      type: Object as PropType<UserDTO>,
      default: () => {
        const defaultUser: UserDTO = {
          id: 0,
          email: '',
          first_name: '',
          last_name: '',
          avatar: '',
        };
        return defaultUser;
      },
    },
  },
  computed: {
    fullName(): string {
      return `${this.user.first_name} ${this.user.last_name}`;
    },
  },
});
</script>

<style lang="scss" scoped>
@import '@/assets/styles/theme.scss';

.user-card {
  padding: 1rem;
  margin: 0px 1rem 1rem 0px;
  max-width: 200px;
  border-radius: 8px;
  cursor: pointer;

  transition: all 0.2s linear;

  &:hover {
    background-color: $dark2;
  }

  .avatar {
    border-radius: 8px;
  }
  .title {
    text-transform: uppercase;
    color: $light;
  }
  .subtitle {
    color: $gray;
    cursor: auto;
  }
}
</style>
