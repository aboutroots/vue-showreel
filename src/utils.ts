export const sleep = async (ms: number): Promise<void> => {
  /**
   * Sleep for given number of miliseconds
   */
  // eslint-disable-next-line no-promise-executor-return
  await new Promise((r) => setTimeout(r, ms));
};
