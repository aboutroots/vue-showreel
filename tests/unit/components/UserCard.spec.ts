import UserCard from '@/components/UserCard.vue';
import { shallowMount } from '../utils';

describe('UserCard tests suite', () => {
  test('renders the component successfully', () => {
    const wrapper = shallowMount(UserCard);

    const container = wrapper.find('.UserCard');

    expect(container.exists()).toBeTruthy();
  });

  test('matches snapshot', () => {
    const wrapper = shallowMount(UserCard);
    expect(wrapper.element).toMatchSnapshot();
  });
});
