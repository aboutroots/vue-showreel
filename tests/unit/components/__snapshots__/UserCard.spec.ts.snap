// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard tests suite matches snapshot 1`] = `
<router-link
  class="UserCard user-card"
  to="[object Object]"
>
  <img
    alt="User avatar"
    class="avatar"
    src=""
  />
   
  <h3
    class="title"
  >
     
  </h3>
   
  <p
    class="subtitle"
  >
    
  </p>
</router-link>
`;
